code/modules/awaymissions/mission_code/UO71-terrorspiders.dm

```diff
@@ -0,0 +1,296 @@
+	One test monkey went from perfect health to death in under 30 seconds - from 3 bites.<br>\
+	I am so very glad I'm not responsible for keeping these things contained."
+
+/obj/item/weapon/paper/terrorspiders7
+	name = "paper= 'Security Orders'"
+	info = "<b>Directives for Officer James</b><br>\
+	<br>\
+	1. Find a way to lift the bridge lockdown.<br>\
+	2. Kill any spiders you encounter.<br>\
+	3. Once on the bridge, open a path to the Queen in the south, and find a way to kill her, ending the infestation. <br>\
+	Good luck, Officer.<br>\
+	"
+
+/obj/item/weapon/paper/terrorspiders8
+	name = "paper= 'Last Words'"
+	info = "<b>The Last Words of Nurse Barnes</b><br>\
+	<br>\
+	It is too much. First Phil, then those damn spiders bite Mary too. Heaven help us. <br>\
+	I'll be evacuating as soon as I've grabbed my supplies. I had to pay for those myself. Stupid budget cuts. <br>\
+	The new world they've found for us isn't that hospitable, but at least there are no deadly spid---<br>\
+	(the writing trails off, as if the writer was interrupted)<br>\
+	"
+
+
+/obj/item/weapon/gun/energy/gun/nuclear/awaymission_aeg
+	name = "Wireless Energy Gun"
+	desc = "An energy gun that recharges wirelessly during away missions. Does not work on the main station."
+	icon_state = "nucgun"
+	projectile_type = "/obj/item/projectile/beam"
+	fire_sound = 'sound/weapons/Laser.ogg'
+	var/inawaymission = 1
+	can_flashlight = 0
+	can_charge = 0
+	charge_cost = 1000
+
+/obj/item/weapon/gun/energy/gun/nuclear/awaymission_aeg/process()
+	var/turf/my_loc = get_turf(src)
+	if(my_loc.z == (MAX_Z + 1))
+		if (inawaymission)
+			charge_tick++
+			if(charge_tick < 4) return 0
+			charge_tick = 0
+			if(!power_supply) return 0
+			if((power_supply.charge / power_supply.maxcharge) != 1)
+				power_supply.give(1000)
+				update_icon()
+		else
+			to_chat(get(src, /mob), "<span class='notice'>Your [src] activates, starting to draw power from a nearby wireless power source.</span>")
+			inawaymission = 1
+	else
+		if (inawaymission)
+			to_chat(get(src, /mob), "<span class='danger'>Your [src] deactivates, as it is out of range from its power source.</span>")
+			power_supply.charge = 0
+			inawaymission = 0
+			update_icon()
+
+/obj/item/weapon/gun/energy/gun/nuclear/awaymission_aeg/update_charge()
+	var/ratio = power_supply.charge / power_supply.maxcharge
+	ratio = round(ratio, 0.25) * 100
+	overlays += "nucgun-[ratio]"
+
+/obj/item/weapon/gun/energy/gun/nuclear/awaymission_aeg/update_reactor()
+	if ((power_supply.charge/power_supply.maxcharge) <= 0.5)
+		overlays += "nucgun-light"
+	else
+		overlays += "nucgun-clean"
+
+/obj/item/weapon/gun/energy/gun/nuclear/awaymission_aeg/update_mode()
+	overlays += "nucgun-kill"
+
+
+
+/obj/item/weapon/gun/energy/gun/nuclear/awaymission_aeg/attack_self(mob/living/user as mob)
+	to_chat(user, "<span class='danger'>[name] appears to only have one setting, scrawled hastily on it in pen: 'SPIDERS!!!'. This is probably a bad sign.</span>")
```

Comment:
Wait, so you only want it to be able to shoot lethals? This wasn't clear to me, initially. Now that I know that, it would probably be best just go go with Some generic laser gun that only processes on the z-level--then you don't have all the other special handling the AEG has.